packages/devextreme/js/__internal/ui/chat/chat_message_group.ts

```diff
@@ -63,30 +64,34 @@ class MessageGroup extends Widget<MessageGroupOptions> {
       });
     }
 
-    this._renderMessageGroupInformation(messages?.[0]);
-    this._renderMessageBubbles(messages);
+    this._renderMessageGroupInformation(items?.[0]);
+    this._renderMessageBubbles(items);
   }
 
-  _renderMessageBubbles(messages): void {
-    messages.forEach((message, index) => {
-      const $bubble = $('<div>');
+  _renderMessageBubble(message: Message, index: number, length: number): void {
+    const $bubble = $('<div>');
 
-      const isFirst = index === 0;
-      const isLast = index === messages.length - 1;
+    const isFirst = index === 0;
+    const isLast = index === length - 1;
 
-      if (isFirst) {
-        $bubble.addClass(CHAT_MESSAGE_BUBBLE_FIRST_CLASS);
-      }
+    if (isFirst) {
+      $bubble.addClass(CHAT_MESSAGE_BUBBLE_FIRST_CLASS);
+    }
 
-      if (isLast) {
-        $bubble.addClass(CHAT_MESSAGE_BUBBLE_LAST_CLASS);
-      }
+    if (isLast) {
+      $bubble.addClass(CHAT_MESSAGE_BUBBLE_LAST_CLASS);
+    }
 
-      $bubble.appendTo(this.element());
+    $bubble.appendTo(this.element());
 
-      this._createComponent($bubble, MessageBubble, {
-        text: message.text,
-      });
+    this._createComponent($bubble, MessageBubble, {
+      text: message.text,
+    });
+  }
+
+  _renderMessageBubbles(messages): void {
+    messages.forEach((message, index) => {
+      this._renderMessageBubble(message, index, messages.length);
```

Comment:
messages.length is an immutable value here, we can assign this value to a variable

Reply:
I don't think it is necessary to put such data in a separate variable. This code is easy to read now

Reply:
It's performance optimization. It allows us to avoid recalculating it repeatedly

Reply:
`const array = new Array(10000);
   console.log(array.length);
            console.time('performanceMeasurement');

            const length = array.length;
            for (let j = 0; j < length; j++) {
                const a = 100;
            }

            console.timeEnd('performanceMeasurement');
            
            console.time('performanceMeasurement');

            for (let j = 0; j < array.length; j++) {
                const a = 100;
            }

            console.timeEnd('performanceMeasurement');`
            
On my machine the result are:

performanceMeasurement: 0.0830078125 ms
performanceMeasurement: 0.10595703125 ms